src/Helper/PaymentHelper.php

```diff
@@ -57,38 +183,57 @@ public function webformSubmissionPresave(WebFormSubmissionInterface $submission)
         break;
       }
     }
-    if (NULL === $paymentElement) {
-      return;
-    }
+
+    return [
+      'paymentElement' => $paymentElement,
+      'paymentElementMachineName' => $paymentElementMachineName,
+    ];
+  }
+
+  /**
+   * Hook on webform submission presave, modifying data before submission.
+   *
+   * @return void
+   *   Return
+   */
+  public function webformSubmissionInsert(WebFormSubmissionInterface $submission): void {
+    $logger_context = [
+      'handler_id' => 'os2forms_payment',
+      'channel' => 'webform_submission',
+      'webform_submission' => $submission,
+      'operation' => 'submission queued',
+    ];
 
     /*
      * The paymentReferenceField is not a part of the form submission,
      * so we get it from the POST payload.
      */
     $request = $this->requestStack->getCurrentRequest();
-    $paymentReferenceField = $request->request->get('os2forms_payment_reference_field');
+    $paymentId = $request->request->get('os2forms_payment_reference_field');
 
-    /*
-     * The goal here is to store the payment_id, amount_to_pay and posting
-     * as a JSON object in the os2forms_payment submission value.
-     */
-    $submissionData = $submission->getData();
-    $amountToPay = $this->getAmountToPay($submissionData, $paymentElement['#amount_to_pay']);
+    /** @var \Drupal\advancedqueue\Entity\Queue $queue */
+    $queue = $this->getQueue();
+    if (!$queue) {
+      throw new \RuntimeException(sprintf('Queue with ID %s does not exist.', $this->queueId));
```

Comment:
Consider adding and using custom exceptions in your module (see https://github.com/itk-dev/ai-screening/tree/develop/web/modules/custom/ai_screening/src/Exception for an example). Having custom exception in a module can make some exception handling easier (in the future).